### DISCUSSION TITLE
Expected typed value in useController

### DISCUSSION THREAD

**Discussion by inrumi** [react-hook-form/react-hook-form#9107] [Q&A]
Hello everyone! is there any way to tell the user of a reusable component what type of value he is going to receive? like the reusable component sent an object instead of a primitive value like this:

```tsx
const { control, handleSubmit } = useForm({
    defaultValues: {
      phone: {
        number: '',
        country: '',
        code: '',
      },
    },
  });
```

I've already checked out the latest version which includes FieldPathByValue but it only shows an error about an expected type but not what type.

here is an example of what im talking about: https://codesandbox.io/s/react-hook-form-v7-ts-template-forked-7k78tl?file=/src/App.tsx

I'm not sure if this has been mentioned before I've found some threads on github related to `FieldArrays` and `useController` (which is what i'm using here) but not is not exactly what I'm asking ?

**Comment by GRA0007** [comment]
I'm having this same issue. All our custom form components return objects, not strings as values, and we want that to be fully typed in the form. It worked fine wrapping every component in a `<Controller>`, as we just made sure the `value` and `onChange` of our custom fields used the correct types, but we're trying to transition to `useController` so we don't have to wrap every field in Controller, and it seems there's no way for the field component itself to define what type it's value must be?

**Reply by leapful** [reply]
@GRA0007 

Please provide a codesandbox url to get more details about your issue. Normally, you can pass specific type value to a generic Field component that is matched with `form.control` to make it type safe.
